### PR TITLE
fix readme

### PR DESCRIPTION
**Pull request by zargot**
Mainly readme formatting. See commits for details.

The only actual change is to not allow strings as rule-ids. If there is a need for this, then please discuss.

### CODE REVIEW

**Comment by zargot** [comment]
what are your thoughts on the rule-id type? Is it good enough with an integer for now? @DougManuel @mathew-thomson

**Comment by zargot** [comment]
@mathew-thomson, I talked to yulric about the column-sequences, and we think it's best to wait with them until the next version since there are some possible complications with it, and we don't know if it's that important for the first version. What do you think?

**Comment by DougManuel** [comment]
@zargot what do you mean by column-sequences? 

Are you talking about how a filter from one table could be used to filter a second table? For example, we filter all measures for 2024. Then, identify all sampleID for those measures in 2024. Then, share all rows from a sample table that correspond to the list of sampleIDs.

If so, this feature can wait until the next version, but it would be helpful to ensure our current specification can be extended to include this feature. The above is an example that is a common practice.

**Comment by zargot** [comment]
> @zargot what do you mean by column-sequences?

My apologies for not being clear. The readme specifies how you can select a sequence of columns by writing `a:z` (which would select everything from `a` to `z`, as an alternative to manually specifying each column.

**Comment by mathew-thomson** [comment]
@zargot - sorry for the slow response on these. 

> what are your thoughts on the rule-id type? Is it good enough with an integer for now? @DougManuel @mathew-thomson

I think this is fine, yeah. I only had other string-type ruleIDs in there because they were in the original specification, but I really don't see this as a particularly important use-case. Integers work perfectly well.

> @mathew-thomson, I talked to yulric about the column-sequences, and we think its best to wait with them until the next version since there are some possible complications with it, and we don't know if it's that important for the first version. What do you think?

@DougManuel - I think Joakim is just meaning specifying a series of columns for select by connecting them with a ":". Like column2:column5.
I also don't think that this is a major use case - I think it could be nice because it saves time when defining a rule, and matches with the dplyr `filter` and `select` syntaxes. But I think it can be left to a later version.

**Review by mathew-thomson — CHANGES_REQUESTED** [review]
Looks good! Looks to be mostly formatting changes, but also caught my mistakes in one of the example tables, haha. My only real additional comment is that I suppose we'll need to remove the line about multiple columns, but that's very minor.

**Review by yulric — APPROVED** [review]
PR looks good. [One comment regarding making ruleId a number](https://github.com/Big-Life-Lab/PHES-ODM-sharing/pull/7#issuecomment-2061559755).

**Comment by zargot** [comment]
> One issue with making ruleIds strictly numbers is that copying pasting rules between sheets will most likely result in collisions, they would have to be updated in the new sheet. ...

I don't think it'll be an issue for users to update the rule id, since it's a very small task and they would probably want to change it anyway to make it part of their own system.

A custom name for the rule could still be kept in the notes-column, in case they want names as well.

I would remove rule-ids all together if group-rules didn't need to reference other rules by id, so I think it's good to avoid adding additional features or workflows that depend on them.

**Comment by mathew-thomson** [comment]
> My only real additional comment is that I suppose we'll need to remove the line about multiple columns, but that's very minor.

@zargot - Sorry I was a little spaced out on cold meds. I was thinking I was making a comment about the need to delete to text about sequential columns, but I know that I saw you already deleted it. So you can disregard this comment. Apologies!

**Comment by mathew-thomson** [comment]
> One issue with making ruleIds strictly numbers is that copying pasting rules between sheets will most likely result in collisions, they would have to be updated in the new sheet.

@yulric - a fair comment, and I see where you're coming from. I think, though, that this would be a very rare case. Given that all the rules for all orgs are defined in one sheet, I don't foresee users needing to have or maintain multiple rules sheets, so copying and pasting between sheets shouldn't be all that common. As far as copying rules from other folks, I also agree with @zargot on this one that folks will need to be following the rules they put in closely and modifying them, so I hope they would be paying close enough attention to avoid issues like this. I agree that we can be open to exploring it in successive version, though!

**Comment by zargot** [comment]
> > My only real additional comment is that I suppose we'll need to remove the line about multiple columns, but that's very minor.
> 
> @zargot - Sorry I was a little spaced out on cold meds. I was thinking I was making a comment about the need to delete to text about sequential columns, but I know that I saw you already deleted it. So you can disregard this comment. Apologies!

I didn't delete it, but I saw you commented that you would delete it. Do you still wanna do it? :smile:

File: README.md [hunk 1]
```diff
@@ -86,105 +176,156 @@ Some examples are given below:
...
+-   The **ruleId** column is mandatory for all rules and each value is unique
+    across the entire sheet (`sharing.csv`). It must be a number.
 
 #### 3.2. Filtering Rows
 
-Once the columns and tables for inclusion have been specified, users can specify which rows should be shared using rules with the `filter` mode. Note that rules that filter can use values in any columns, including columns that are not being shared in the final output. To specify a `filter` rule, users need to specify the table or tables in the `table` column, and define the `mode` as filter. Then users can specify the columns which the filter will act on in the `key` column, specify the nature of the filter using the `operator` column, and the filter values in the `value` column. The general structure for the filter argument is:
-
-```         
-  **column name** **operator** **value**
```
Comment (DougManuel) [inline README.md:134]: This isn't rendering or displaying on my build.

**Review by DougManuel — APPROVED** [review]
The document looks very good.

I made edits to the introduction and then just one or two typos further down.

I suggest keeping rule ID as numbers, and it is helpful to have rule-id as a reference (beyond 'groups') for user versioning, referencing, discussions, and what have you.

We are opinionated in parts of the document when we say, `share.csv`, but the actual name of the file should be up the user. And I think of the file name as the namespace. For the rule ID. I discussed earlier whether we want to be able to combine schema together, but that could be the next version. Meaning, you create a basic share.csv template, and then "import" or reference that template in a new, second schema. So, no cutting and pasting.


Finally, it occurred to me that select statement could get verbose if there are many headers in a table that a custodian wants to share. This could be addressed in a next versions, with approaches to streamline and support this process such as in R dpylr that uses the following:

image

File: README.md (hunk 1 above)
Comment (zargot) [reply README.md:134]: good catch. fixed now.

**Comment by zargot** [comment]
> I made edits to the introduction and then just one or two typos further down.

I'm not able to see any commits, did you push them yet?

> Finally, it occurred to me that select statement could get verbose if there are many headers in a table that a custodian wants to share. This could be addressed in a next versions, ...

I agree that we can improve it in the next version, but we may want to hear from the users first. They may want to select half of the columns, or they may want all except one. Each need requires a different solution, and I don't know how many of them we want to support. Let's see.

**Comment by zargot** [comment]
I'm merging now. I've removed the parts mentioning sequential columns. Feel free to make a new PR if I missed anything.

I'm not sure about using ':' for date-ranges, etc. I was thinking to use ';' for everything since the operator already provides the context. Let's discuss this at the next meeting or something, @mathew-thomson.

**Pull request merged**